lib/omnibus/software.rb

```diff
@@ -328,6 +328,46 @@ def default_version(val = NULL)
     expose :default_version
 
     #
+    # Set or retrieve the {#license} of the software to build.
+    #
+    # @example
+    #   license 'Apache 2.0'
+    #
+    # @param [String] val
+    #   the license to set for the software.
+    #
+    # @return [String]
+    #
+    def license(val = NULL)
+      if null?(val)
+        @license || 'Unspecified'
+      else
+        @license = val
+      end
+    end
+    expose :license
+
+    #
+    # Set or retrieve the location of a {#license_file}
+    # of the software.  It can either be a relative path inside
+    # the source package or a URL.
+    #
+    #
+    # @example
+    #   license_file 'LICENSES/artistic.txt'
+    #
+    # @param [String] val
+    #   the location of the license file for the software.
+    #
+    # @return [String]
+    #
+    def license_file(file)
+      license_files << file
+      license_files.dup
+    end
+    expose :license_file
```

Comment:
Would it be more appropriate for the DSL to expose a block based function like `version`:

```ruby
license 'Apache 2.0' do
  file 'LICENSE'
end

license 'BSD' do
  file 'BSDL'
  file 'COPYING'
  file 'LEGAL'
end
```

This would obviously require a `Omnibus::License` class to instance_eval against but it could easily be used with both `Project` and `Software`.